### PR TITLE
Add Highlight feature for the selected video on the 'Individual Courses Page' #252

### LINKED ISSUE #252: Add Highlight feature for the selected video on the 'Individual Courses Page'
## Is your feature request related to a problem? Please describe.

In the Individual Courses page there is no indication regarding the video which is currently selected . Even after clicking on the component there is no change .

before1

## Describe the solution you'd like

I would like to add a method which detects the currently selected video and highlights the ListItem as shown below .
after1

### PR DESCRIPTION
**Pull request by samad-yar-khan**
**ISSUE 252**

fixes #252 

**Describe the changes you've made**

- I have added a 'videoPlaying' property to our state on the the individualCourses page .
- This property will take the ID of the current video selected/played but for now the index of the video from the list is being stored here as it is same as the key for the component as of now .
- Initial value of the 'videoPlaying' property is set to -1 for now , indicating no video selected but later this can be changed to the last video being played by the user.
- Pressing/Clicking on a video from the List on the individualCourses page triggers a playVideo function which takes the key/index of the video which has been selected and resets the state by setting the 'videoPlaying' property equal to the selected videos key/index.
- At the time of rendering of Videos List items in the List , if the video list item being rendered is the one which is being played , then a separate 'playVideo' class and style is added to it using ternary operator and conditional rendering .

**Describe if there is any unusual behavior (Any Warning) of your code(Write `NA` if there isn't)**

 Unusual Behaviour : `NA`

**Additional context (OPTIONAL)**

Page after adding the highlight feature .
Here the first video is selected .

**Checklist**

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] The title of my pull request is a short description of the requested changes.

**Provide a Deployed link of route/page that needs to review**
Preview: http://localhost:3000/mycourses/:coursename

<!-- Please add deployed link to here text >

### CODE REVIEW

**Comment by samad-yar-khan** [comment]
@Abhishek-kumar09 Sorry for the delay . Can you please review my pull request ?

**Comment by samad-yar-khan** [comment]
@Abhishek-kumar09 Sir , Can you please review my pull request  ?

File: src/pages/CoursePageViewWithVideo/IndividualCourse.js [hunk 1]
```diff
@@ -98,6 +103,10 @@ export default function CoursePageViewWithVideo(props) {
     setState({ ...state, open: false });
   };
 
+  const playVideo = (index) => {
+    setState({ ...state, vidPlaying: index });
```
Comment (Abhishek-kumar09) [inline src/pages/CoursePageViewWithVideo/IndividualCourse.js:107]: you are using `vidPlaying` here, but `videoPlaying` in the state on line 92

**Comment by samad-yar-khan** [comment]
@Abhishek-kumar09 My bad , I changed the property name in the last but didn't make changes in the places where I used it. Thankyou for pointing this out .  Changed all occurrences of vidPlaying property to videoPlaying in the latest commit . Seems to be working fine . Can you please review once again ?

File: src/pages/CoursePageViewWithVideo/IndividualCourse.js [hunk 2]
```diff
@@ -172,7 +181,11 @@ export default function CoursePageViewWithVideo(props) {
           <List style={{ margin: 0, padding: 0 }}>
             {[1, 2, 3, 4, 5].map((item, index) => {
               return (
-                <ListItem key={index}>
+                <ListItem
```
Comment (Abhishek-kumar09) [inline src/pages/CoursePageViewWithVideo/IndividualCourse.js:184]: Since it is clickable, add `cursor: pointer` to the list items.

**Comment by samad-yar-khan** [comment]
@Abhishek-kumar09 I have changed the cursor to pointer for the List Items .

File: src/pages/CoursePageViewWithVideo/IndividualCourse.js [hunk 3]
```diff
@@ -85,6 +89,7 @@ export default function CoursePageViewWithVideo(props) {
     open: false,
     vertical: "top",
     horizontal: "center",
+    videoPlaying: -1,
```
Comment (Abhishek-kumar09) [inline src/pages/CoursePageViewWithVideo/IndividualCourse.js:92]: change it to `playingIndex`

**Comment by samad-yar-khan** [comment]
@Abhishek-kumar09  I have made the changes . Have a look and let me know if somethings missing/needs to be changes .

**Review by Abhishek-kumar09 — APPROVED** [review]

**Comment by Abhishek-kumar09** [comment]
Thanks @samad-yar-khan

**Comment by samad-yar-khan** [comment]
@Abhishek-kumar09  Thanks a lot for reviewing my code multiple times and suggesting all the changes  :)

**Pull request merged**